src/net/ifaddrs/mod.rs

```diff
@@ -76,17 +76,43 @@ use self::iff_flags::IffFlags;
 mod sockaddr;
 use self::sockaddr::{IfAddrValue, sockaddr_to_ifaddrvalue};
 
-pub type InterfaceMap = HashMap<String, Vec<InterfaceAddr>>;
+mod tests;
+
+pub type InterfaceMap<'a> = HashMap<String, Vec<InterfaceAddr<'a>>>;
 
 /// Represents a handle into the operating system's knowledge about network
 /// interfaces present on the system. Allows the user to iterate over
 /// interface configurations.
-pub struct InterfaceAddrs {
+pub struct InterfaceAddrs<'a> {
     inner: *mut libc::ifaddrs,
-    current: Option<&'static libc::ifaddrs>,
+    current: Option<&'a libc::ifaddrs>,
+    do_free: bool,
```

Comment:
It looks like `do_free` only exists to support your test code.  You can eliminate it by using `mem::forget` in the test code to prevent `drop` from being run.